src/mock.rs

```diff
@@ -0,0 +1,239 @@
+    testing::{Digest, DigestItem, Header},
+    traits::{BlakeTwo256, IdentityLookup}, //OnFinalize, OnInitialize},
+    BuildStorage,
+};
+
+/// Module which has a full Substrate module for 
+/// mocking behaviour of MembershipRegistry
+pub mod registry {
+
+    use srml_support::*;
+    use super::*;
+
+    #[derive(Encode, Decode, Default, Clone, PartialEq, Eq)]
+    pub struct Member<AccountId> {
+        pub id: AccountId
+    }
+
+    decl_storage! {
+        trait Store for Module<T: Trait> as MockForumUserRegistry {
+
+            pub ForumUserById get(forum_user_by_id) config(): map T::AccountId => Member<T::AccountId>;
+
+        }
+
+        // A weird hack required to convince decl_storage that our field above
+        // does indeed use the T, otherwise would start complaining about phantom fields.
+        // An alternative fix is to add this extra key when initiatilising a gensisconfig for this module
+        // _genesis_phantom_data: Default::default() 
+        extra_genesis_skip_phantom_data_field;
+    }
+
+    decl_module! {
+        pub struct Module<T: Trait> for enum Call where origin: T::Origin {}
+    }
+
+    impl<T: Trait> Module<T> {
+
+        pub fn get_member(account_id: &T::AccountId) -> Option<Member<T::AccountId>> { 
+            
+            if <ForumUserById<T>>::exists(account_id) {
+                Some(<ForumUserById<T>>::get(account_id))
+            } else {
+                None
+            }
+
+        }
+
+        pub fn add_member(member: & Member<T::AccountId>) {
+            <ForumUserById<T>>::insert(member.id.clone(), member.clone());
+        }
+
+    }
+
+    impl<T: Trait> ForumUserRegistry<T::AccountId> for Module<T> {
+
+        fn get_forum_user(id: &T::AccountId) -> Option<ForumUser<T::AccountId>> {
+
+            if <ForumUserById<T>>::exists(id) {
+
+                let m = <ForumUserById<T>>::get(id);
+
+                Some(ForumUser{
+                    id : m.id 
+                })
+
+            } else {
+                None
+            }
+
+        }
+    }
+
+    pub type TestMembershipRegistryModule = Module<Runtime>;
+
+}
+
+impl_outer_origin! {
+    pub enum Origin for Runtime {}
+}
+
+// Workaround for https://github.com/rust-lang/rust/issues/26925 . Remove when sorted.
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct Runtime;
+
+impl system::Trait for Runtime {
+    type Origin = Origin;
+    type Index = u64;
+    type BlockNumber = u64;
+    type Hash = H256;
+    type Hashing = BlakeTwo256;
+    type Digest = Digest;
+    type AccountId = u64;
+    type Lookup = IdentityLookup<Self::AccountId>;
+    type Header = Header;
+    type Event = ();
+    type Log = DigestItem;
+}
+
+impl timestamp::Trait for Runtime {
+    type Moment = u64;
+    type OnTimestampSet = ();
+}
+
+impl Trait for Runtime {
+    type Event = ();
+    type MembershipRegistry = registry::TestMembershipRegistryModule;
+}
+
+// Here are a few testing utilities and fixtures, will reorganize
```

Comment:
These fixtures are not used inside mock.rs, as they have been moved to tests.rs

Reply:
Oh, thanks for catching this!